### ISSUE TITLE
Join links behavior for public Topics - no login?

### ISSUE THREAD

**Issue by tiblu** [citizenos/citizenos-fe-old#405]
**Current**

* User generates a /join link (for ex: https://app.citizenos.com/join/jTMPwgeZ) for a PUBLIC topic
* Another User opens the link
* User is redirected to Citizen OS login
* After login, Topic is added to the Users Topic list.

**What should we do differently?**

* Maybe we should not require Users to log in for public Topics?
* We could show "join discussion" for every PUBLIC topic, clicking it will ask User to log in and then add the Topic to Users list?
* IF User does log in being on the /join page, we could then add the Topic to his/her list?
* .... any other ideas?

Source: Slack #product 06 Feb 2020 5:34 PM EEST

**Comment by loorm** [comment]
Triage 17: We see the problem. Need to discuss, how this should work. We have no "request access" functionality right now. Otherwise, QR code could redirect to public topic link. User could read the topic, but would not be able to "join" the topic.

**Comment by loorm** [comment]
Linked to #311

**Comment by loorm** [comment]
Moving to triage, because I think this issue can be closed without further actions. Reasons:

- If user is not logged in and simply browses around in public topics and finds one, where they would like to participate, there is a link saying "You need to log in to participate". That's essentially the same as "Request access", because once you log in, then for public topics, you can now join the discussion. I don't really see a user case, where I randomly find a public topic and would like to join in co-creation.

- If user is given a join link (no matter whether the topic is public or private), then they follow the link, log in and get the access, that is assigned for the join link (once #311 is completed).

- If topic authors want to share a link for public topics, that people can simply follow to have a look at and read the topic without participation, then they can use the social share (once #311 is complete).

To sum it up, completing issue #311 and currently existing functionality combined solves the issue and nothing further is needed. Therefore, this issue can be closed.

**Comment by anettlinno** [comment]
Triage 47. Successfully replicated. For keeping the scope of #311 issue, we will solve this issue separately. Est. dev. time 3 days. Sending to development.

**Comment by ilmartyrk** [comment]
Needs verification that automated test checks all the possible conditions for user join and group/user permission levels

**Comment by tiblu** [comment]
Agreed with @anettlinno  that in the scope of this task we will implement following solution:

* When User opens up a /join link for a PUBLIC Topic, User is redirected to the Topics public URL.

WHEN we implement https://github.com/citizenos/citizenos-fe/issues/289 a User can then request access to edit the Topic.

**Comment by tiblu** [comment]
**DONE**

**Comment by anettlinno** [comment]
Tested. Sharing public topic join link in FB and LinkedIn works fine.
